CMakeLists.txt

```diff
@@ -27,29 +28,130 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
+set(_SPM_SAVED_BUILD_TESTING ${BUILD_TESTING})
+set(_SPM_SAVED_BUILD_EXAMPLES ${BUILD_EXAMPLES})
+set(_SPM_VENDOR_DEPENDENCIES)
+
+set(BUILD_EXAMPLES NO)
+set(BUILD_TESTING NO)
+
+find_package(ArgumentParser CONFIG)
+if(NOT ArgumentParser_FOUND)
+  message("-- Vending swift-argument-parser")
+  FetchContent_Declare(ArgumentParser
+    GIT_REPOSITORY https://github.com/apple/swift-argument-parser
+    GIT_TAG 1.2.3)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ArgumentParser)
+endif()
+
+find_package(TSC CONFIG)
+if(NOT TSC_FOUND)
+  message("-- Vending swift-tools-support-core")
+  FetchContent_Declare(ToolsSupportCore
+    GIT_REPOSITORY https://github.com/apple/swift-tools-support-core
+    GIT_TAG main)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ToolsSupportCore)
+endif()
 
-  find_package(LLBuild CONFIG)
-  if(NOT LLBuild_FOUND)
+find_package(LLBuild CONFIG)
+if(NOT LLBuild_FOUND)
+  if(APPLE)
     find_package(LLBuild REQUIRED)
+  else()
+    message("-- Vending swift-llbuild")
+    set(LLBUILD_SUPPORT_BINDINGS Swift)
+    FetchContent_Declare(LLBuild
+      GIT_REPOSITORY https://github.com/apple/swift-llbuild
+      GIT_TAG main)
+    list(APPEND _SPM_VENDOR_DEPENDENCIES LLBuild)
   endif()
+endif()
 
-  find_package(ArgumentParser CONFIG REQUIRED)
-  find_package(SwiftDriver CONFIG REQUIRED)
-  find_package(SwiftCollections CONFIG REQUIRED)
-  find_package(SwiftASN1 CONFIG REQUIRED)
-  find_package(SwiftCertificates CONFIG REQUIRED)
-  find_package(SwiftCrypto CONFIG REQUIRED)
+find_package(SwiftASN1 CONFIG)
+if(NOT SwiftASN1_FOUND)
+  message("-- Vending swift-asn1")
+  FetchContent_Declare(ASN1
+    GIT_REPOSITORY https://github.com/apple/swift-asn1
+    GIT_TAG 1.1.0)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ASN1)
+endif()
+
+find_package(SwiftCertificates CONFIG)
+if(NOT SwiftCertificates_FOUND)
+  message("-- Vending swift-certificates")
+  FetchContent_Declare(Certificates
+    # GIT_REPOSITORY https://github.com/apple/swift-certificates
+    # GIT_TAG 1.1.0)
+    GIT_REPOSITORY https://github.com/compnerd/swift-certificates
+    GIT_TAG vendor)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES Certificates)
+endif()
+
+find_package(SwiftCollections CONFIG)
+if(NOT SwiftCollections_FOUND)
+  message("-- Vending swift-collections")
+  FetchContent_Declare(Collections
+    GIT_REPOSITORY https://github.com/apple/swift-collections
+    GIT_TAG release/1.0)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES Collections)
+endif()
+
+find_package(SwiftCrypto CONFIG)
+if(NOT SwiftCyrpto_FOUND)
+  message("-- Vending swift-crypto")
+  FetchContent_Declare(Crypto
+    GIT_REPOSITORY https://github.com/apple/swift-crypto
+    GIT_TAG 3.0.0)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES Crypto)
+endif()
+
+find_package(SwiftDriver CONFIG)
+if(NOT SwiftDriver_FOUND)
+  message("-- Vending swift-driver")
+  FetchContent_Declare(Driver
+    GIT_REPOSITORY https://github.com/apple/swift-driver
+    GIT_TAG main)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES Driver)
```

Comment:
What about dependencies of driver, e.g. yams?

Reply:
swift-driver is responsible for its dependencies, and will bring those in.